samtranslator/model/eventsources/push.py

```diff
@@ -921,7 +927,36 @@ def _add_swagger_integration(  # type: ignore[no-untyped-def] # noqa: too-many-s
                 path=self.Path, method_name=self.Method, request_parameters=parameters
             )
 
-        api["DefinitionBody"] = editor.swagger
+        if merge_definitions:
+            api["DefinitionBody"] = self._get_merged_definitions(api["DefinitionBody"], editor.swagger)
+        else:
+            api["DefinitionBody"] = editor.swagger
+
+    def _get_merged_definitions(
+        self, source_definition_body: Dict[str, Any], dest_definition_body: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        """
+        Merge SAM generated swagger definition(dest_definition_body) into inline DefinitionBody(source_definition_body):
+        - for a conflicting key, use SAM generated value
+        - otherwise include key-value pairs from both definitions
+        """
+        merged_definition_body = source_definition_body.copy()
+        source_body_paths = merged_definition_body.get("paths", {})
+
+        try:
+            path_method_body = dict_deep_get(source_body_paths, [self.Path, self.Method]) or {}
+        except InvalidValueType as e:
+            raise InvalidResourceException(self.logical_id, f"Property 'DefinitionBody' is invalid: {str(e)}") from e
+
+        generated_path_method_body = dest_definition_body["paths"][self.Path][self.Method]
+        # this guarantees that the merged definition use SAM generated value for a conflicting key
+        merged_path_method_body = {**path_method_body, **generated_path_method_body}
```

Comment:
if `path_method_body` is not a dict, will it be a dip?

Reply:
no, exception is caught.

Reply:
Had to add another check to make sure no dip.